File: e2e/scenarios/geoservices_error_surfacing/probes/probe.py
#!/usr/bin/env python3
"""Require the frozen Python SDK to surface a real GeoServices error envelope."""
import os
import sys

SERVER = os.environ["HONUA_SERVER_URL"].rstrip("/")

try:
    from honua_sdk import HonuaClient, HonuaHttpError
except Exception as exc:  # noqa: BLE001
    print(f"SKIP: honua-sdk not importable: {exc}")
    sys.exit(2)

try:
    with HonuaClient(SERVER) as client:
        result = client.query_features("__honua_parity_missing__", 0, where="1=1))")
except HonuaHttpError as exc:
    print(f"PASS: Python SDK raised HonuaHttpError on 200+{{error}}: {exc}")
    sys.exit(0)
except Exception as exc:  # noqa: BLE001
    print(f"FAIL: Python SDK raised the wrong exception: {type(exc).__name__}: {exc}")
    sys.exit(1)

print(f"FAIL: Python SDK returned success on 200+{{error}}: {result!r}")
sys.exit(1)

File: e2e/scenarios/geoservices_error_surfacing/probes/probe.mjs
// Require the frozen JavaScript SDK to surface a real GeoServices error envelope.
const SERVER = (process.env.HONUA_SERVER_URL || "").replace(/\/$/, "");

let sdk;
try {
  sdk = await import("@honua/sdk-js");
} catch (error) {
  console.log(`SKIP: @honua/sdk-js not importable: ${error}`);
  process.exit(2);
}

try {
  const client = new sdk.HonuaClient({ baseUrl: SERVER });
  const result = await client.queryFeatures({
    serviceId: "test",
    layerId: 0,
    where: "__release_missing_field__=1",
    method: "GET",
  });
  console.log(`FAIL: JavaScript SDK returned success on 200+{error}: ${JSON.stringify(result)}`);
  process.exit(1);
} catch (error) {
  if (error instanceof sdk.HonuaHttpError) {
    console.log(`PASS: JavaScript SDK raised HonuaHttpError on 200+{error}: ${error}`);
    process.exit(0);
  }
  console.log(`FAIL: JavaScript SDK raised the wrong error type: ${error}`);
  process.exit(1);
}
